flink-ml-servable-core/src/main/java/org/apache/flink/ml/linalg/Vector.java

```diff
@@ -24,29 +24,29 @@
 
 import java.io.Serializable;
 
-/** A vector of double values. */
+/** A vector representation of numbers. */
 @TypeInfo(VectorTypeInfoFactory.class)
 @PublicEvolving
-public interface Vector extends Serializable {
+public interface Vector<K extends Number, V extends Number> extends Serializable {
 
     /** Gets the size of the vector. */
-    int size();
+    K size();
 
     /** Gets the value of the ith element. */
-    double get(int i);
+    V get(K i);
 
     /** Sets the value of the ith element. */
-    void set(int i, double value);
+    void set(K i, V value);
 
-    /** Converts the instance to a double array. */
-    double[] toArray();
+    /** Converts the instance to a primitive array. */
+    Object toArray();
```

Comment:
The implementation of this API always return `double[]`. Would it be simpler to still use `double[] toArray()`?